src/corppa/poetry_detection/passim/get_passim_excerpts.py

```diff
@@ -0,0 +1,82 @@
+"""
+Gather excerpts from passim page-level results.
+"""
+
+import argparse
+import csv
+import pathlib
+import sys
+
+import orjsonl
+
+
+def get_passage_matches(input_file):
+    """
+    Extracts and yields the passim-identified passage-level matches
+    """
+    for page in orjsonl.stream(input_file):
+        if not page["n_spans"]:
+            # Skipe pages without matches
+            continue
+        for poem_span in page["poem_spans"]:
+            match = poem_span.copy()
+            match["page_id"] = page["page_id"]
+            yield match
+
+
+def save_passim_passage_matches(input_file, output_file):
+    with open(output_file, mode="w", newline="") as tsvfile:
+        fieldnames = [
+            "page_id",
+            "ref_id",
+            "ref_corpus",
+            "page_start",
+            "page_end",
+            "ref_start",
+            "ref_end",
+            "page_excerpt",
+            "ref_excerpt",
+            "aligned_page_excerpt",
+            "aligned_ref_excerpt",
+        ]
+        writer = csv.DictWriter(tsvfile, fieldnames=fieldnames, dialect="excel-tab")
+        writer.writeheader()
+        for match in get_passage_matches(input_file):
+            writer.writerow(match)
+
+
+def main():
+    """
+    Command-line access to build TSV file of passage-level passim matches.
```

Comment:
how well does tsv work for multi-line text?

if we keep this, in future I'd like to see an option for csv output also

Reply:
Perfectly fine. Tabs are better for all the command-line utilities, and arguably tabs are less common than commas in the texts we're processing.

I can switch to CSV for the sole reason that Google Docs has some broken exporting happen around TSVs (this is a short coming of that tool only)